modtools/intro-loader.lua

```diff
@@ -82,63 +82,64 @@ if args.help then
     return
 end
 
-local inputfolder = dfhack.getDFPath() .. "\\data\\announcement\\"
-if args.folder then
-    inputfolder = dfhack.getDFPath() ..  args.folder .. "\\"
-end
-
 local prefix = "INTRO"
 if args.file then
     prefix = args.file
 end
 
-if args.rand then
-    math.randomseed(os.time())
+local inputfolder = dfhack.getDFPath() .. "/data/announcement/"
+if args.folder then
+    inputfolder = dfhack.getDFPath() ..  args.folder .. "/"
 end
 
-
 --Build current index of files in directory
 do
-    local OS = dfhack.getOSType()
-    if OS == 'windows' then
-        os.execute("dir \"" .. inputfolder .. "\" /L /B >index.txt")
-    elseif OS == 'linux' or OS == 'darwin' then --apple's real name is darwin, apparently
-        os.execute("ls \"" .. inputfolder .. "\" -1 -A >index.txt")
-    end
-    
-    local list = io.open(dfhack.getDFPath() .. "\\data\\announcement\\index.txt","r")
-    indextext = list:read("*all")
-    io.close(list)
+    local index = io.open(dfhack.getDFPath() .. "/data/announcement/index.txt","w")
+	local files = dfhack.filesystem.listdir(inputfolder)
+    for foo,filename in pairs(files) do
+		index:write(filename .. "\n")
+	end
+    io.close(index)
```

Comment:
Oh my brain derped on that, since I had to do it the hard way earlier and that needed to be way separated from everything else just to make the document sensible. Notably, the filenames are actually in the second 'column' from listdir, oddly enough. That was confounding until I actually looked into index.txt and saw it was just full of numbers.